app/Http/Controllers/Settings/MultiFAController.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace App\Http\Controllers\Settings;
+
+use Google2FA;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Foundation\Auth\RedirectsUsers;
+use PragmaRX\Google2FALaravel\Support\Authenticator;
+
+class MultiFAController extends Controller
+{
+    use RedirectsUsers;
+
+    protected $redirectTo = '/settings/security';
+
+    /**
+     * Session var name to store secret code.
+     */
+    private const SESSION_TFA_SECRET = '2FA_secret';
+
+    /**
+     * Create a new authentication controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('web');
+    }
+
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function enableTwoFactor(Request $request)
+    {
+        //generate new secret
+        $secret = $this->generateSecret();
+
+        $user = $request->user();
+
+        //generate image for QR barcode
+        $imageDataUri = Google2FA::getQRCodeInline(
```

Comment:
Not sure if it's an issue, but static analysis says that this is an instance method, called statically

Reply:
You're right, we should use `app('pragmarx.google2fa')->getQRCodeInline` instead.
Thanks !
Do you want to open an issue ?

Reply:
Or just add it to the (great) PR #878

Reply:
Have added